objects/boat/Boat.gd

```diff
@@ -0,0 +1,8 @@
+class_name Boat
+extends StaticBody2D
+
+var _can_enter = false
```

Comment:
Is it for multiplayer purpose? The boat is opened when the chest is delivered?
BTW: starting a variable name with a underscore is to make it private but you change and get it from `Player`.

Reply:
You can keep the "Can Enter" mechanism even if we aren't multiplayer-ready yet